src/libstd/sys/unix/backtrace.rs

```diff
@@ -133,7 +133,17 @@ pub fn write(w: &mut Write) -> io::Result<()> {
     result::fold(iter, (), |_, _| ())
 }
 
-#[cfg(not(all(target_os = "ios", target_arch = "arm")))]
+#[cfg(target_os = "nacl")]
+#[inline(never)]
+pub fn write(_w: &mut Write) -> io::Result<()> {
+    use io::ErrorKind;
+    Err(io::Error::new(ErrorKind::PermissionDenied,
+                       "can't read the stack instruction pointer \
+                        to find the backtrace"))
+}
+
+#[cfg(not(any(all(target_os = "ios", target_arch = "arm"),
+              target_os = "nacl")))]
```

Comment:
er, hit comment too soon...

Basically with an organization like that there shouldn't need to be spaghetti imports and each implementation is basically in its own files. Then the only #[cfg] should be at the top level selecting the implementation to use.